### PR TITLE
feat: tax code matcher ui

### PR DESCRIPTION
**Pull request by peelar**
Zrzut ekranu 2023-06-20 o 11 13 35
Zrzut ekranu 2023-06-20 o 11 13 28

## Context
This PR contains the UI for the Tax Code Matcher feature. The PR for the backend is here: https://github.com/saleor/apps/pull/564.

Here is the context around some choices that I've made here:

Tax Code Matcher is not tied to a particular provider connection. You can only configure it once per provider, per all channels.

The current structure of the app revolves around the idea of a provider connection: an instance of provider configuration. The Tax Code Matcher doesn't necessarily fit into that structure, so we had to make some deliberate decisions to make it fit.

We decided to create a separate tax code matcher section on the homepage. It checks if there is any connection for each provider and displays buttons for configuring the provider tax code matcher (so "Avatax button" and "TaxJar button").

In the provider tax code matcher view, we fetch all the providers, take the first one, and use it for authenticating the request for the tax codes list.

### Tax codes:

- `AvataxClientTaxCodeService` - because the tax codes returned from Avatax require some refinement.
- `AvataxTaxCodeMatcherTable` and `TaxJarTaxCodeMatcherTable` - for displaying & assigning the tax class - tax code pairs.
- `MatcherSection` and `MatcherTable` - checks if there is at least one provider and renders a button for tax code matcher.

### Other:
- `_Select` - wrapper for Select because [long labels don't render properly](https://github.com/saleor/macaw-ui/issues/477).
- `AppDashboardLink` - a reimplementation of `ui/TextLink` because it doesn't handle my case. I can add it in a separate ticket.

## Todo
- [x] - fix bug when the value doesn't preselect when `router.push` to the page
- [ ] - QA

## Checklist
* [ ] `.github/dependabot.yaml` is up-to date.
* [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

**Comment by krzysztofwolski** [comment]
@peelar : Re matcher view: have you considered using a bit different style, to use whole width of the screen for the mapping? 
We had a similar view in the Search app - two column view with really long names of categories. Since it's single-purpose view, the description can be on top and then you have more space for the table.

File: apps/taxes/src/modules/ui/app-dashboard-link.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export const AppDashboardLink = ({
+  children,
+  href,
+  ...rest
+}: PropsWithChildren<{ href: string }>) => {
+  const appBridge = useAppBridge();
+
+  const redirectToDashboardPath = () => {
+    appBridge.appBridge?.dispatch(actions.Redirect({ to: href }));
```
Comment (lkostrowski) [inline apps/taxes/src/modules/ui/app-dashboard-link.tsx:17]: We have TextLink in packages/ui

**Review by lkostrowski — COMMENTED** [review]
so far so good

File: apps/taxes/src/modules/avatax/ui/avatax-tax-code-matcher-table.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  React.useEffect(() => {
+    if (avataxMatches) {
+      const match = avataxMatches?.find((item) => item.data.saleorTaxClassId === taxClassId);
+
+      if (match) {
+        setValue(match.data.avataxTaxCode);
+      }
+    }
+  }, [avataxMatches, taxClassId]);
```
Comment (krzysztofwolski) [inline apps/taxes/src/modules/avatax/ui/avatax-tax-code-matcher-table.tsx:24]:
```suggestion
  const { data: avataxMatches, isLoading: isMatchesLoading } =
    trpcClient.avataxMatches.getAll.useQuery({
    onSuccess: (data) => {
      if(!data){
        return;
      }
      const match = data.find((item) => item.data.saleorTaxClassId === taxClassId);

      if (match) {
        setValue(match.data.avataxTaxCode);
      }
    }
    });
```

Not _wrong_, personally I find using onSuccess/error provided by react query more ergonomic to use. I see you used it in mutation

File: apps/taxes/src/modules/taxjar/ui/taxjar-tax-code-matcher-table.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import { AppCard } from "../../ui/app-card";
+
+const SelectTaxCode = ({ taxClassId }: { taxClassId: string }) => {
+  const [value, setValue] = React.useState("");
+  const { notifySuccess, notifyError } = useDashboardNotification();
+
+  const { data: taxJarMatches, isLoading: isMatchesLoading } =
+    trpcClient.taxJarMatches.getAll.useQuery();
+
+  React.useEffect(() => {
```
Comment (krzysztofwolski) [inline apps/taxes/src/modules/taxjar/ui/taxjar-tax-code-matcher-table.tsx:16]: As previous comment regarding query.

**Review by lkostrowski — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**